analytic_engine/src/table/version.rs

```diff
@@ -727,7 +727,9 @@ impl TableVersion {
         picker_ctx: PickerContext,
         picker: &CompactionPickerRef,
     ) -> picker::Result<CompactionTask> {
-        let inner = self.inner.read().unwrap();
+        // Pick will set FileHandle to being_compacted state, so we require
+        // write lock here to prevent other threads pick same file to compact.
+        let inner = self.inner.write().unwrap();
 
         picker.pick_compaction(picker_ctx, &inner.levels_controller)
```

Comment:
"levels_controller: &LevelsController" -> 
"levels_controller: &mut LevelsController"

Force to use wirte lock here?

Reply:
Good suggestion.